app/services/scenes-collections/nodes/overlays/text.ts

```diff
@@ -34,16 +34,39 @@ export class TextNode extends Node<ISchema, IContext> {
     this.data = { settings };
   }
 
+  checkTextSourceFace(settings: any): Promise<void> {
```

Comment:
Rather than passing it in and mutating it (considered bad form), I would just mutate the settings directly on `this.data.settings`.  It's the same object.